gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -797,8 +797,13 @@ OGRMultiPolygon *OGRFlatGeobufLayer::readMultiPolygon(const Feature *feature, ui
 OGRGeometry *OGRFlatGeobufLayer::readGeometry(const Feature *feature)
 {
     auto pXy = feature->xy();
-    if (pXy == nullptr)
-        return nullptr;
+    auto pZ = feature->z();
+    auto pM = feature->m();
+    VALIDATE_POINTER1(pXy, "readGeometry", nullptr);
```

Comment:
in a debug build, this will issue a CE_Fatal error, which will abort the process. So this macro is only to be used to detect programming errors, not corrupt data

Reply:
Ah, I see. Replaced with standard nullptr checks.